packages/amplify-graphql-api-construct-tests/src/__tests__/deploy-velocity/deploy-velocity-test-core.ts

```diff
@@ -0,0 +1,216 @@
+      wasDestroyed = true;
+    });
+  });
+};
+
+/**
+ * Generate n fake uuids for testing
+ * @param count the number of ids to return
+ * @returns the list of fake ids
+ */
+export const generateFakeUUIDs = (count: number): string[] => Array.apply(0, new Array(count)).map((_) => faker.string.uuid());
+
+/**
+ * Given a list of object, and a chunk size, break into a list of lists, all of which has,
+ * at most, the number of elements provided by chunkSize.
+ * @param inputArray the input elements to partition
+ * @param chunkSize the max size of any given partition for the resulting arrays.
+ * @returns the input array partitioned by chunksize.
+ */
+export const splitArray = <T>(inputArray: T[], chunkSize: number): T[][] => {
+  if (chunkSize < 1) {
+    throw new Error('Chunk size must be greater than 0');
+  }
+  return inputArray.reduce((resultArray, item, index) => {
+    const chunkIndex = Math.floor(index / chunkSize);
+
+    if (!resultArray[chunkIndex]) {
+      resultArray[chunkIndex] = []; // start a new chunk
+    }
+
+    resultArray[chunkIndex].push(item);
+
+    return resultArray;
+  }, [] as T[][]);
+};
+
+const generateApiPostProcessorFile = (postProcessor: ApiPostProcessor): string => `module.exports = ${postProcessor.toString()}`;
+
+const getRecordCount = async (endpointConfig: EndpointConfig): Promise<number> => {
+  let nextToken = null;
+  let resultCount = 0;
+  do {
+    const response = await validateGraphql({
+      ...endpointConfig,
+      query: /* GraphQL */ `
+        query LIST_TODOS {
+          listTodos(nextToken: ${nextToken ? `"${nextToken}"` : 'null'}) {
+            items { id }
+            nextToken
+          }
+        }
+      `,
+      expectedStatusCode: 200,
+    });
+    nextToken = response.body.data.listTodos.nextToken;
+    resultCount += response.body.data.listTodos.items.length;
+  } while (nextToken);
+  return resultCount;
+};
+
+export const recordCountDataProvider = (recordCount: number, mutationBuilder: (uuid: string, i: number) => string) => {
+  return async (endpointConfig: EndpointConfig): Promise<void> => {
+    // Generate Data in batches of 50 per request
+    const mutationBatches = splitArray(generateFakeUUIDs(recordCount), 50).map((uuidBatch) => uuidBatch.map(mutationBuilder).join('/n'));
+
+    // And execute up to 10 requests in parallel
+    for (const mutationBatch of splitArray(mutationBatches, 10)) {
+      await Promise.all(
+        mutationBatch.map((mutations) =>
+          validateGraphql({
+            ...endpointConfig,
+            query: /* GraphQL */ `mutation CREATE_TODOS { ${mutations} }`,
+            expectedStatusCode: 200,
+          }),
+        ),
+      );
+    }
+  };
+};
+
+export const recordCountDataValidator = (expectedRecordCount: number) => {
+  return async (endpointConfig: EndpointConfig): Promise<void> => {
+    const recordCount = await getRecordCount(endpointConfig);
+    expect(recordCount).toEqual(expectedRecordCount);
+  };
+};
+
```

Comment:
If I understand correctly, the two functions below are used only for single record use case

Reply:
That's correct. The 5 `single-record` tests.